temporal/server_test.go

```diff
@@ -37,16 +43,89 @@ import (
 	"go.temporal.io/server/tests/testutils"
 )
 
-// TestNewServer verifies that NewServer doesn't cause any fx errors
+// TestNewServer verifies that NewServer doesn't cause any fx errors, and that there are no unexpected error logs after
+// running for a few seconds.
 func TestNewServer(t *testing.T) {
+	t.Parallel()
+
+	cfg := loadConfig(t)
+	logger := newErrorLogDetector(t)
+
+	server, err := NewServer(
+		WithConfig(cfg),
+		WithLogger(logger),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, server.Stop())
+	})
+	require.NoError(t, server.Start())
+	time.Sleep(10 * time.Second)
+}
+
+func loadConfig(t *testing.T) *config.Config {
+	cfg := loadSQLiteConfig(t)
+	setTestPorts(cfg)
+	return cfg
+}
+
+// loadSQLiteConfig loads the config for the sqlite persistence store. We use sqlite because it doesn't require any
+// external dependencies, so it's easy to run this test in isolation.
+func loadSQLiteConfig(t *testing.T) *config.Config {
 	configDir := path.Join(testutils.GetRepoRootDirectory(), "config")
 	cfg, err := config.LoadConfig("development-sqlite", configDir, "")
 	require.NoError(t, err)
+
 	cfg.DynamicConfigClient.Filepath = path.Join(configDir, "dynamicconfig", "development-sql.yaml")
-	_, err = NewServer(
-		ForServices(DefaultServices),
-		WithConfig(cfg),
-	)
-	assert.NoError(t, err)
-	// TODO: add tests for Server.Run(), etc.
+	return cfg
+}
+
+// setTestPorts sets the ports of all services to something different from the default ports, so that we can run the
+// tests in parallel.
+func setTestPorts(cfg *config.Config) {
+	port := 10000
+
+	for k, v := range cfg.Services {
+		rpc := v.RPC
+		rpc.GRPCPort = port
+		port++
+		rpc.MembershipPort = port
+		port++
+		v.RPC = rpc
+		cfg.Services[k] = v
+	}
+}
+
+// newErrorLogDetector returns a logger that fails the test if it logs any errors or warnings, except for the ones that
+// are expected. Ideally, there are no unexpected errors or warnings, but we still want this test to avoid introducing
+// any new ones while we are working on removing the existing ones.
+func newErrorLogDetector(t *testing.T) *log.MockLogger {
+	ctrl := gomock.NewController(t)
```

Comment:
nice [ab]use of gomock :)

can you create the controller in the test and pass it in? and call Finish? just to match existing patterns, I know it won't really do anything

Reply:
ctrl.Finish() is not needed anymore

Reply:
😉 done